html/ops/db_update.php

```diff
@@ -1162,6 +1162,12 @@ function update_5_9_2018() {
     ");
 }
 
+function update_8_23_2018() {
+    $retval = do_query("alter table host add index host_cpid (host_cpid)");
+    return $retval && do_query("alter table host add index host_domain_name (domain_name)");
```

Comment:
I understand that it is a common practice in this script to add indexes this way but why don't we check index existence before adding it? I'm not very familiar with MySql but it's can be done easily in MSSQL and I'm pretty sure that there is a way to do the same here.

Reply:
So I looked into this and the easiest way to do this is simply to check the errno returned if the create failed and if it is 1061 then it means that the key name is a duplicate.  I was going to implement this but then I started thinking through different scenarious about what could be going on and I came up with the following:

1) The project created their own index that is named the same.  In this case stopping processing and having the project take a look at what is going on is the right thing.
2) The project applied the index early by seeing a needed index and adding it.  In that case it would be ok to skip adding it.

I didn't come up with other scenarios.   Given that these two scenarios suggest different behavior with regards to the appropriate action to take I'm not sure how to improve on the existing behavior.

The script does inform the user of the following in the event of a failure:
```
performing update update_8_23_2018
Doing query:
alter table host add index host_cpid (host_cpid)
Failed:
Duplicate key name 'host_cpid'
WARNING: database upgrade failed.
MySQL error message: Duplicate key name 'host_cpid'
Please find the update queries in html/ops/db_update.php
and perform them manually.
When done, edit PROJECT_DIR/db_revision so that it contains the line 27027
```
which provides a pretty straight forward instructions on how to respond.

thoughts?

Reply:
Hm, what happens if there's already an index for the column in question but under a different name? Never tried that, I think...

Reply:
@TheAspens in case you missed it, there's still this open question of mine 😏

Reply:
MariaDB happily adds a index with a different name even if it duplicates an existing index